spec/matchers/make_allocations_matcher.rb

```diff
@@ -9,10 +9,10 @@
     end
 
     @allocations = stats.allocations.to_a.size
-    @allocations == expected
+    @allocations <= expected
```

Comment:
This makes the matcher more resilient against optimizations in future Ruby versions (which shouldn't fail the build).